src/idsk/components/subscription-form/subscription-form.js

```diff
@@ -0,0 +1,50 @@
+import '../../../govuk/vendor/polyfills/Function/prototype/bind'
+import '../../../govuk/vendor/polyfills/Event' // addEventListener and event.target normaliziation
+import { nodeListForEach } from '../../../../package/govuk/common'
+
+function SubscriptionForm ($module) {
+  this.$module = $module
+}
+
+SubscriptionForm.prototype.init = function () {
+  // Check for module
+  var $module = this.$module
+  if (!$module) {
+    return
+  }
+
+  // button to toggle content
+  var $forms = $module.querySelectorAll('form')
+
+  nodeListForEach($forms, function ($form) {
+    $form.addEventListener('submit', this.handleSubmitForm.bind(this))
+  }.bind(this))
+}
+
+/**
+ * An event handler for submit event on $form
+ * @param {object} e
+ */
+SubscriptionForm.prototype.handleSubmitForm = function (e) {
+  var $el = e.target || e.srcElement
+  e.preventDefault()
+
+  // check if email is set
+  if (this.$module.querySelector('input[type=email]').value === '')
+    return
+
+  // hide form and agreement text
+  $el.style.display = 'none'
+  this.$module.querySelector('.agreement-text').style.display = 'none'
+
+  // change texts to sent state
+  var $elementsToChange = this.$module.querySelectorAll('.description-text, .title-text')
+  nodeListForEach($elementsToChange, function ($element) {
+    $element.innerHTML = $element.dataset.sentState
+  }.bind(this))
```

Comment:
cisto teoreticky: mas na stranke 2 krat subscription form. Jeden mas niekde v tele druhy v paticke... vyplnis form v tele odosles. Skryje ti form a zobrazi dakujeme spravu na oboch. Skor by som asi chcel zobrazit success spravu len na forme ktoty som vyplnil nie vsetky

Reply:
Spravil som to trošku inak.. Ale každopádne, tu sa o to stará ten "this.module", ktorý vyberie elementy iba v tom konkrétnom komponente. Či to chápem zle?
![image](https://user-images.githubusercontent.com/31508588/144832728-8bee5386-4552-4d93-a547-531a2018a932.png)

Reply:
pockaj, teeraz nesom si 100% isty. Rozsireny priklad je iny priklad podla mna tam su iframe cize tu by to malo byt ok. Ale skus is na pageu dat dva krat tento modul ci sa bude spravat korektne. Ak ano je zbytocne v init nodeListForEach na forms lebo tam moze byt len jeden. Dufam ze sa chapeme :)

Reply:
Hodil som to na random stránku a tam to tiež funguje v pohode. A pravda s tým forEach.. Len mam pocit, že som videl inde, že sa to tiež robí takto iba pre jeden element, tak ja že to je nejaká good practic :D
Ale tak sa mi to zdalo asi len
![image](https://user-images.githubusercontent.com/31508588/144842331-26cee659-7893-4e98-9e84-d30bddf83d13.png)

Reply:
Moze byt, ze to niekde tak je.. bohuzial nekonzistentnost a veci s tym spojene su castokrat dosledkom nedostatku casu.